DigitalLearningSolutions.Data/Services/RegistrationService.cs

```diff
@@ -162,33 +159,58 @@ public void PromoteDelegateToAdmin(AdminRoles adminRoles, int categoryId, int de
 
             var adminUser = userDataService.GetAdminUserByEmailAddress(delegateUser.EmailAddress);
 
-            if (adminUser != null)
+            if (adminUser?.Active is false && adminUser.CentreId == delegateUser.CentreId)
             {
-                throw new AdminCreationFailedException(AdminCreationError.EmailAlreadyInUse);
+                userDataService.ActivateAdmin(adminUser.Id);
+                userDataService.UpdateAdminUser(
+                    delegateUser.FirstName,
+                    delegateUser.LastName,
+                    delegateUser.EmailAddress,
+                    delegateUser.ProfileImage,
+                    adminUser.Id
+                );
+                passwordDataService.SetPasswordByAdminId(adminUser.Id, delegateUser.Password);
+                userDataService.UpdateAdminUserPermissions(

```

Comment:
Worth noting that this does not force the IsCentreManager permission to false like we do when registering a new admin.  So if a user was a centre manager, and was then deactivated, then reactivated, they would still be a centre manager.

I don't _think_ this will cause any issues, but it is a scenario where we have the potential for an admin to "create" an admin with higher permissions than they themselves have. @SteveJacksonSoft shall I leave this as it is, or ensure the IsCentreManager permission is set to false here?

Reply:
I've made the ReactivateAdmin data service method revoke permissions higher than CentreAdmin (CentreManager and UserAdmin but UserAdmin is unlikely to be true in any case).

Though I now realise that the promote to admin button and controller require the user to be a centre manager anyway. So centre managers are probably safe reactivating other centre managers. Given that centre manager is a substantial permission, we might still want to revoke it (and UserAdmin) by default. Not sure if there are limits on the number of centre managers, but I don't think centre managers are something admins can easily promote others to. @SteveJacksonSoft  thoughts?